packages/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncherPlugin.java

```diff
@@ -55,9 +55,7 @@ public void onMethodCall(MethodCall call, Result result) {
         launchIntent = new Intent(Intent.ACTION_VIEW);
         launchIntent.setData(Uri.parse(url));
       }
-      if (mRegistrar.activity() == null) {
-        launchIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-      }
+      launchIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
This sounds like the wrong thing to do since if I want to launch a URL, I might want to append this activity to the task of my application. NEW_TASK will always force a new task to be created which will mess up the back button behavior (if I am reading the documentation correctly: https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_NEW_TASK). Could you please clarify why you want to do this?

Reply:
```
/io.flutter.plugins.urllauncherexample E/MethodChannel#plugins.flutter.io/url_launcher: Failed to handle method call
    android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
        at android.app.ContextImpl.startActivity(ContextImpl.java:1376)
        at android.app.ContextImpl.startActivity(ContextImpl.java:1363)
        at android.content.ContextWrapper.startActivity(ContextWrapper.java:323)
        at io.flutter.plugins.urllauncher.UrlLauncherPlugin.onMethodCall(UrlLauncherPlugin.java:61)
        at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:200)
        at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:163)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:148)
        at android.os.Looper.loop(Looper.java:151)
        at android.app.ActivityThread.main(ActivityThread.java:5692)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
```

Because the context if not an Activity context, so it will encounter above error.
I check the git history and find this is an error introduced by [this commit](https://github.com/flutter/plugins/commit/f057a22423e7fd858deb2c01bc5cdd2dcf5c91a1#diff-4aad027b9fcffd3e0f439689b391c85d) .I am going to update my pull request.

Reply:
@mehmetf @timtraversy PR has updated.

Reply:
> In other words, don't change line 63 and verify that you can close the webview opened via activity.launchIntent() 

I have checked that it works

Reply:
Here's what I meant:

```java
    } else if (call.method.equals("launch")) {
      Intent launchIntent;
      boolean useWebView = call.argument("useWebView");
      boolean enableJavaScript = call.argument("enableJavaScript");
      Activity activity = mRegistrar.activity();
      if (activity == null) {
         result.error("Launching a URL requires a foreground activity.");
         return;
      }
      if (useWebView) {
        launchIntent = new Intent(context, WebViewActivity.class);
        launchIntent.putExtra("url", url);
        launchIntent.putExtra("enableJavaScript", enableJavaScript);
      } else {
        launchIntent = new Intent(Intent.ACTION_VIEW);
        launchIntent.setData(Uri.parse(url));
      }
      activity.startActivity(launchIntent);
      result.success(null);
    } else if (call.method.equals("closeWebView")) {
      Intent intent = new Intent("close");
      mRegistrar.context().sendBroadcast(intent);
      result.success(null);
    } else {
```

and please make sure it works.